functions/graphql/schema/directives/auth.js

```diff
@@ -50,15 +68,18 @@ class AuthDirective extends SchemaDirectiveVisitor {
         const [data, queryArgs, context] = args;
         const { user } = context;
 
-        console.log("DATA", data);
-
+        // QUESTION: since this ensureFieldsWrapped function is called before this property is set,
```

Comment:
I'm not entirely clear on what this question is asking, but check the above answer.

Reply:
From the docs:

>What makes this example tricky is that the OBJECT version of the directive needs to wrap all fields of the object, even though some of those fields may be individually wrapped by @auth directives at the FIELD_DEFINITION level, and we would prefer not to rewrap resolvers if we can help it.
>One drawback of this approach is that it does not guarantee fields will be wrapped if they are added to the schema after AuthDirective is applied, and the whole getUser(context.headers.authToken) is a made-up API that would need to be fleshed out. In other words, we’ve glossed over some of the details that would be required for a production-ready implementation of this directive, though we hope the basic structure shown here inspires you to find clever solutions to the remaining problems.